packages/destination-actions/src/destinations/dawn/identifyUser/index.ts

```diff
@@ -0,0 +1,54 @@
+import type { ActionDefinition, RequestClient } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import { DawnIdentifyUser } from '../dawn-types'
+
+const getEventFromPayload = (payload: Payload): DawnIdentifyUser => {
+  const identifyUserPayload: DawnIdentifyUser = {
+    user_id: payload.user_id || '',
+    traits: payload.traits || {}
+  }
+  return identifyUserPayload
+}
+
+const processData = async (request: RequestClient, settings: Settings, payload: Payload[]) => {
+  const identifyUsers = payload.map((value) => getEventFromPayload(value))
+  return request('https://api.dawnai.com/segment-identify', {
+    method: 'post',
+    json: identifyUsers,
+    headers: {
+      authorization: `Bearer ${settings.writeKey}`
+    }
+  })
+}
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Identify User',
+  description: '',
+  fields: {
+    user_id: {
+      label: 'User ID',
+      type: 'string',
+      description: 'The user ID performing the event.',
+      required: true,
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    traits: {
+      label: 'Traits',
+      type: 'object',
+      description: 'The traits of the user.',
+      required: false,
+      default: {
+        '@path': '$event' // Currently dumping the entire event object as traits
+      }
+    }
```

Comment:
```suggestion
    traits: {
      label: 'Traits',
      type: 'object',
      description: 'The traits of the user.',
      required: false,
      default: {
        '@path': '$.traits'
      }
    }
```

If you want to let the user provide the ability to map traits, then I would change the traits field to default to '$.traits'.

If you want to be able to ingest the entire Segment payload, then I would either:

Add a new field for the entire event:
```
eventPayload: {
      label: 'Segment Event Payload',
      description: 'The entire Segment payload.',
      type: 'object',
      readOnly: true,
      required: true,
      default: {
        '@path': '$event'
      }
    }
```

Alternatively, you can access the entire event payload from the perform block by restructuring rawData in the second argument:
```
  perform: async (request, { settings, payload, rawData }) => {
    return processData(request, settings, [payload])
  }
```